docs/api/files.md

```diff
@@ -0,0 +1,77 @@
+# Files
+
+> This page is incomplete, stay tuned.
+
+## Upload file
+
+### Code samples
+
+```shell
+# Example request using curl
+curl -X POST http://coder-server:8080/api/v2/files \
+  -H 'Accept: application/json' \
+  -H 'Content-Type: application/x-tar' \
+  -H 'Coder-Session-Token: API_KEY'
+```
+
+`POST /files`
+
+Notice: Swagger 2.0 doesn't support file upload with a `content-type` different than `application/x-www-form-urlencoded`.
```

Comment:
This information seems distracting for anyone reading this document. Perhaps it's only relevant for the self-hosted variant with the API test fields?

Reply:
The truth is that we need the definition for `POST /files` in the `swagger.json` to render docs, but we know that the Swagger UI will not work due to the content-type limitation. It might be relevant for Swagger users to know that it is not supported. I don't know what is the best notification option for this issue.

Maybe we should hide it from docs (docs postprocessing) and just leave it in the Swagger UI?